src/ios/PushPlugin.m

```diff
@@ -119,7 +119,16 @@ - (void)unregister:(CDVInvokedUrlCommand*)command;
             NSLog(@"unsubscribe from topic: %@", topic);
             [pubSub unsubscribeFromTopic:topic];
         }
-    } else {
```

Comment:
Instead of adding an `else if` what about putting that code inside the `else` statement? So…

```
    else {
        if ([self usesFCM]) {
            [[FIRInstanceID instanceID] deleteIDWithHandler:^void(NSError *_Nullable error){
                if (error) {
                    [self failWithMessage:command.callbackId withMsg:@"" withError:error];
                } else {
                    [self successWithMessage:command.callbackId withMsg:@"unregistered"];
                }
            }];
        }
        [[UIApplication sharedApplication] unregisterForRemoteNotifications];
        [self successWithMessage:command.callbackId withMsg:@"unregistered"];
    }
```

This way the APNS token will get released as well.

Reply:
I tested it and there is a small problem. Success callback gets called even when FCM token delete failed. So I did something like this,
```
} else if ([self usesFCM]){
        [[FIRInstanceID instanceID] deleteIDWithHandler:^void(NSError *_Nullable error){
            if (error) {
                [self failWithMessage:command.callbackId withMsg:@"" withError:error];
            } else {
                [[UIApplication sharedApplication] unregisterForRemoteNotifications];
                [self successWithMessage:command.callbackId withMsg:@"unregistered"];
            }
        }];
    }
    else {
        [[UIApplication sharedApplication] unregisterForRemoteNotifications];
        [self successWithMessage:command.callbackId withMsg:@"unregistered"];
    }
```

Reply:
@macdonst ,
Are the changes ok?